infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServerCache.java

```diff
@@ -167,15 +126,25 @@ public long size() {
      * @param newTail        global tail
      */
     public void put(ConflictTxStream conflictStream, long newTail) {
-        conflictCache.put(conflictStream, newTail);
+        if (cacheHash.size() == cacheSize) {
```

Comment:
I don't understand this logic. Can you please explain?

Reply:
@xnull This means when the cache is full, you put must first evict (make space in the cache), before you can add the entry.